app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -266,6 +268,8 @@ def parse_instance(instance)
 
         rg_ems_ref = get_resource_group_ems_ref(instance)
 
+        labels = parse_labels(instance.tags)
```

Comment:
These are new right, you didn't collect them before?
Terminology note:
in container provider we called them "labels" in code and used section `labels` because [that's what Kubernetes calls them](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/).
IIUC in AWS and Azure they original concept is called tags?
It gets confusing if we collect "tags" into "labels" and map those into MIQ "tags" :-(

I propose code following UI.  Are you going to display azure tags in UI, when looking at a VM?  How would you call them?
Even if you don't display them, IIRC mapping add/edit UI currently speaks of "label name", "label values" and that needs to be rethought.

- IMHO "single pane of glass" consistency is not a relevant consideration here, if "label" won't mean anything to a user using MIQ with only Azure.
- I can see 2 things called "tags" being confusing as well, but IMHO "Azure tags" vs "ManageIQ/Cloudforms tags" (or "Smart Management tags" if we must™, ugh) could work.

Reply:
I believe it is better to stay in line with Amazon's code patterns here to ease further DRYing (I'm not going to DRY it right now, because it should be backported).
Maybe with the https://github.com/ManageIQ/manageiq-providers-azure/pull/229 PR (which is not suggested for backporting), or with a fully separate one.
What about the UI: I'm not sure how it should be presented there, so I'm aligning to the Amazon's side again for simplicity.